internal/command/command_service.go

```diff
@@ -172,6 +174,36 @@ func (cs *CommandService) UpdateDataPlaneHealth(ctx context.Context, instanceHea
 	return err
 }
 
+func (cs *CommandService) SendDataPlaneResponse(ctx context.Context, response *mpi.DataPlaneResponse) error {
+	slog.DebugContext(ctx, "Sending data plane response", "response", response)
+
+	backOffCtx, backoffCancel := context.WithTimeout(ctx, cs.agentConfig.Common.MaxElapsedTime)
+	defer backoffCancel()
+
+	sendDataPlaneResponse := func() error {
+		cs.subscribeClientMutex.Lock()
+		defer cs.subscribeClientMutex.Unlock()
+
+		if cs.subscribeClient == nil {
+			return errors.New("subscribe client is not initialized")
+		}
+
+		err := cs.subscribeClient.Send(response)
+		if err != nil {
+			slog.ErrorContext(ctx, "Failed to send data plane response", "error", err)
+
+			return err
+		}
+
+		return nil
+	}
```

Comment:
This can be pulled out into its own function: https://github.com/nginx/agent/blob/5f289815d11e5e6ef9852e25982db13abc202763/internal/command/command_service.go#L250-L272

Reply:
Merged Donals PR should be changed now